### PR TITLE
Add/update alt text in documentation

### PR DESCRIPTION
**Pull request by isabela-pf**
This PR adds or updates existing [alt text](https://moz.com/learn/seo/alt-text) to Project Jupyter documentation. It includes alt text for
- all images on the main documentation
- all images used in community call notes

If there are images in this repo you think were missed, I'd love to know so we can address them in the future.

If you have any questions about reviewing alt text, let me know and I'd be happy to tell you what I look for.

Thanks in advance for reviewing this PR! 🌻 

---

This alt text was contributed by wonderful volunteers at one of the [Jupyter accessibility workshops](https://blog.jupyter.org/join-us-for-the-jupyter-accessibility-workshops-part-1-133e0e522d1b)! They teamed up to make progress on so many images. 

If you are curious how we have a commit authored by ten people (wow! 🎉), feel free to check out our [working PR](https://github.com/isabela-pf/jupyter/pull/1) or the [How to Participate section of the event agenda](https://github.com/Quansight-Labs/jupyter-accessibility-workshops/blob/main/events/2022-january-22/mentored_sprint_agenda.md#how-to-participate-in-writing-alt-text).

### CODE REVIEW

File: docs/source/projects/architecture/content-architecture.rst [hunk 1]
```diff
@@ -49,6 +49,7 @@ between the frontends and the IPython Kernel is described in
 The core execution machinery for the kernel is shared with terminal IPython:
 
 .. image:: figs/ipy_kernel_and_terminal.png
+   :alt: Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ.
```
Comment (choldgraf) [inline docs/source/projects/architecture/content-architecture.rst:52]: Is there a best practice for including information as alt text when that information would *also* be useful as "regular" text that comes just before or after the image? Some of these alt-texts actually read like captions which would be useful regardless of whether the reader is using assistive technology. If we defined these as captions instead of `alt-text` (or used very simple alt-text in addition to captions) would that be an option from an a11y perspective?

Comment (isabela-pf) [reply docs/source/projects/architecture/content-architecture.rst:52]: This is a great question! I've been having this discussion with a few different people recently, but I'm not aware of any best practices written for this yet since what is or isn't  included *outside* of the image is often considered the realm of the documentation's writing style guide (or lack thereof). 

There are, however, resources on how to combine captions and alt text that we could use if we go that direction. Basically, if we'd rather have this as body text or a caption and not alt text, we want to have an empty `alt` element for the images (which means they are skipped). Here's the [W3C recommendation on images with adjacent text](https://www.w3.org/WAI/tutorials/images/decorative/#image-with-adjacent-text-alternative).

There's also been some discussion around whether we want to use [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) to indicate that there is an image while explicitly linking to its explanation. This could be helpful if the related information is not directly next to the image. I personally think this could be a misuse of the tag, though, especially when our goal is to use proper HTML over adding ARIA. 

cc @tonyfast, @gabalafou, and @ohrely (in case you all are interested)

---

Overall, I'm open to making some of these alt text contributions captions of some kind in this or a future PR. My main goal is to give these sprint contributors credit (via commit) however we decide to do so.

Comment (choldgraf) [reply docs/source/projects/architecture/content-architecture.rst:52]: Hmmm - here's an idea: what if we converted these into figures with captions, instead of images. [Here's an example of what they would look like from Jupyter Book](https://jupyterbook.org/content/figures.html?highlight=figures#figures).

There's a directive for this in Sphinx. Basically, this image directive:

```rst
.. image:: figs/ipy_kernel_and_terminal.png
   :alt: Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ. 
```

would become this figure directive:

```rst
.. figure:: figs/ipy_kernel_and_terminal.png
   
   Venn diagram showing that Terminal IPython and the IPython Kernel both share the same Python executable. Terminal IPython communicates with the Python executable using stdin and stdout, whereas the Python Kernel uses JSON messages sent over ZeroMQ. 
```

File: docs/source/contributing/dev-contributions/contrib_guide.rst [hunk 2]
```diff
@@ -120,7 +120,9 @@ including partial or in-progress work, is appreciated.
 
 Contribution Workflow
 ---------------------
+[Link to full alt-text of Contribution Workflow flow chart.](https://gist.githubusercontent.com/MarsBarLee/2dc5e9c83adf0555d08277b2262302e5/raw/be6a0c46f58d7135a75df7747698cef676b9cdc0/Jupyter%2520Documentation:%2520Contribution%2520Workflow%2520Alt-Text)
```
Comment (MarsBarLee) [inline docs/source/contributing/dev-contributions/contrib_guide.rst:123]: For this alt-text, I provided both short alt-text and long alt-text. In this context, longer alt-text is valuable as short-alt text would be excluding essential information in this flowchart. I have linked out to the long alt-text.

During the workshop, I linked to a personal gist due to the ease of use and speed. However, for submission into the Jupyter documentation, I think linking to a markdown file in the Jupyter documentation would be better. 

Where would a markdown file with an image's long text be contained? In this case, the file is in contributing/dev-contributions but the image is in _static/_images. I'm leaning towards putting it in the images folder. The markdown file could be titled 'contribution_workflow_long_alt_text.md' so it would show next to the image 'contribution_workflow.png'.

In reference to [WCAG](https://www.w3.org/WAI/tutorials/images/complex/#a-text-link-to-the-long-description-adjacent-to-the-image) and [Deque](https://dequeuniversity.com/tips/long-descriptions), both refer to linking out to long alt-text, but do not advice where the text should be hosted.

I could create a separate PR after this PR submitted to change the link and add the long alt-text file.

Comment (choldgraf) [reply docs/source/contributing/dev-contributions/contrib_guide.rst:123]: Hmmm - what if we provided this as a footnote? Here's the reStructuredText way to write footnotes in Sphinx: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes

File: docs/source/projects/architecture/content-architecture.rst (hunk 1 above)
Comment (choldgraf) [reply docs/source/projects/architecture/content-architecture.rst:52]: Here's an example from the Jupyter Book docs:

https://jupyterbook.org/en/stable/content/figures.html#figures

Here's a sample HTML snippet:

```html
<div class="figure align-default" id="directive-fig">
<a class="reference internal image-reference" href="../_images/C-3PO_droid.png"><img alt="../_images/C-3PO_droid.png" src="../_images/C-3PO_droid.png" style="height: 150px;"></a>
<p class="caption"><span class="caption-number">Fig. 4 </span><span class="caption-text">Here is my figure caption!</span><a class="headerlink" href="#directive-fig" title="Permalink to this image">¶</a></p>
</div>
```

If that's not a11y-friendly I'm happy to just go with alt tags

Comment (gabalafou) [reply docs/source/projects/architecture/content-architecture.rst:52]: My recommendation is that we remove the alt text of the diagrams and review them. By remove, I mean set the alt text to empty. By review, I mean we should make sure that the surrounding text describes the diagram. In other words, we should make sure the diagrams do not contain any information that isn't already conveyed by reading the article without the diagrams. I want to be clear that I am only referring to diagram-type images, not all images.

Here's why. 

I did a deep dive into this subject this afternoon. I dug through threads on the global accessibility Slack. I read web pages on the W3C and Gov.UK sites. I also tested with my own hardware using combinations of VoiceOver and NVDA with Safari, Chrome and Edge. 

The main problem is that there are cross-screen-reader compatibility issues. Not all screen readers treat the figure tag the same way. 

The other problem is that my alt-text diagram descriptions typically repeat or paraphrase stuff that's already in the article. So unless we make my descriptions redundant/repetitive for all users, I don't think we should potentially degrade the screen reader experience by making it more repetitive and verbose.

This seems to align with the [UK government's guidance for describing images.](https://www.gov.uk/guidance/content-design/images)

If we agree with this course of action, then I will review the PR and use the suggestion feature to make the changes I that I am recommending.

Comment (choldgraf) [reply docs/source/projects/architecture/content-architecture.rst:52]: I defer to your and others expertise when it comes to the vest practices for accessibility. Though it sounds like some of the changes you suggest would also just be more generally useful for readers, if we are removing duplicate information and such. So I'm extra +1 on that as well.

**Review by isabela-pf — COMMENTED** [review]
I left comments on each change to make it easier to find updated areas and understand my logic. Hope that helps!

File: docs/source/projects/architecture/content-architecture.rst [hunk 3]
```diff
@@ -49,6 +49,7 @@ between the frontends and the IPython Kernel is described in
 The core execution machinery for the kernel is shared with terminal IPython:
 
 .. image:: figs/ipy_kernel_and_terminal.png
+   :alt:
```
[the remark this answers is not in the document]
Comment (isabela-pf) [reply docs/source/projects/architecture/content-architecture.rst:52]: Good thought! I wasn't sure if I was thinking too far ahead, but if we both are then I think it's okay. 😆

[the remark this answers is not in the document]
Comment (isabela-pf) [reply docs/source/projects/architecture/content-architecture.rst:52]: Did you want them added only to the top of each document? As of now I have comments below each empty alt.

File: docs/source/contributing/docs-contributions/doc-new-translation.md [hunk 4]
```diff
@@ -372,8 +373,7 @@ ReadTheDocs will build both the source documentation site as well as sites for a
 languages. ReadTheDocs will associate the sites with one another and make them accessible via
 language links in a popup.
 
-![ReadTheDocs popup](static/translation-rtd-popup.png "Screenshot of the ReadTheDocs popup that
-allows a user to view documentation in a different language")
+![Project Jupyter documenation site with the ReadTheDocs popup open on the right, showing links to the documentation in other languages.](static/translation-rtd-popup.png)
```
Comment (gabalafou) [inline docs/source/contributing/docs-contributions/doc-new-translation.md:376]:
```suggestion
![Project Jupyter documentation site with the ReadTheDocs popup open on the right, showing links to the documentation in other languages.](static/translation-rtd-popup.png)
```

File: docs/source/contributing/docs-contributions/repo-structure.md [hunk 5]
```diff
@@ -16,7 +16,7 @@
 - ``makefile`` : used by Sphinx to build the docs
 - ``environment.yml`` : conda build instructions
 
-![``docs`` directory](static/docs-directory.png "Screenshot of docs directory")
+![The docs directory in the jupyter kernel_gateway repo ( path: jupyter/kernel_gateway/docs). The first two elements in the directory are folders that have blue boxes surrounding them: `_static` and `source`.  The third and forth elements in the directory are files with orange boxes surrounding them: `Makefile` and `environment.yml`. There are two more files in the directory, which have no visual emphasis.](static/docs-directory.png)
```
Comment (gabalafou) [inline docs/source/contributing/docs-contributions/repo-structure.md:19]:
```suggestion
![The docs directory in the jupyter kernel_gateway repo (path: jupyter/kernel_gateway/docs). The first two elements in the directory are folders that have blue boxes surrounding them: `_static` and `source`.  The third and forth elements in the directory are files with orange boxes surrounding them: `Makefile` and `environment.yml`. There are two more files in the directory, which have no visual emphasis.](static/docs-directory.png)
```

File: docs/source/projects/architecture/content-architecture.rst [hunk 6]
```diff
@@ -49,6 +49,10 @@ between the frontends and the IPython Kernel is described in
 The core execution machinery for the kernel is shared with terminal IPython:
```
Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:49]: Question about a probably very minor issue: is the fact that there is a colon at the end of the sentence somehow conveyed by the screen reader? In which case, would the document flow for the screen reader user be somewhat disrupted if the phrase preceding the colon is not semantically connected to the phrase following the colon?

In other words, I'm concerned that to a screen reader, we have

> The core execution machinery for the kernel is shared with terminal IPython:
> A kernel process can be connected to more than one frontend simultaneously. In
> this case, the different frontends will have access to the same variables.

But it doesn't make sense to join those two statements with a colon. It makes sense to join-via-colon the first sentence to the image that it describes/captions.

This colon-terminated phrase is really a caption for the image. But the whole figure-caption machinery as of now creates a bad user experience. So I wonder what a good solution would be.

Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:49]:
```suggestion
The core execution machinery for the kernel is shared with terminal IPython.
```

File: docs/source/projects/architecture/content-architecture.rst [hunk 7]
```diff
@@ -66,6 +70,10 @@ the core execution part. Native kernels implement execution and communications
 in the target language:
```
Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:70]:
```suggestion
in the target language.
```

File: docs/source/projects/architecture/content-architecture.rst [hunk 8]
```diff
@@ -118,6 +121,7 @@ as HTML, LaTeX, or reStructuredText. This conversion goes through a series of
 steps:
 
 .. image:: figs/nbconvert.png
+   :alt:
```
[the remark this answers is not in the document]
Comment (gabalafou) [reply docs/source/projects/architecture/content-architecture.rst:130]: agree

File: docs/source/projects/architecture/content-architecture.rst [hunk 9]
```diff
@@ -175,4 +188,4 @@ Below is a high level visual overview of project relationships. It is current as
 
 .. image:: /_static/_images/repos_map.png
    :width: 75%
-   :alt: Architecture diagram of project relationships
+   :alt: Architecture diagram of Jupyter project relationships from servers, applications, API, and kernels.
```
Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:191]: Good lord. How do we make such a dense relationship map accessible? I feel like this is getting into new territory for me

File: docs/source/projects/architecture/content-architecture.rst [hunk 10]
```diff
@@ -102,6 +110,7 @@ to the notebook server, which saves it on disk as a JSON file with a
 ``.ipynb`` extension.
 
 .. image:: figs/notebook_components.png
+   :alt: Communication diagram showing five nodes labeled user, browser, notebook server, notebook file, and kernel. The notebook server is a communication hub. The browser, notebook file and kernel cannot talk to each other directly. They communicate indirectly through the notebook server. The user can only talk to the browser. The notebook server and kernel are colored green whereas the browser is colored purple to draw a distinction between the Jupyter front end versus back end.
```
Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:113]:
```suggestion
   :alt:
   
.. 
   Alt text is intentionally left blank because the image content is described thoroughly in the surrounding text.

The notebook server is a communication hub. The browser, notebook file on disk, and
kernel cannot talk to each other directly. They communicate through the notebook server. 
```

File: docs/source/projects/architecture/content-architecture.rst [hunk 11]
```diff
@@ -102,6 +110,7 @@ to the notebook server, which saves it on disk as a JSON file with a
...
```
Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:114]:
```suggestion
```

Comment (gabalafou) [inline docs/source/projects/architecture/content-architecture.rst:114]:
```suggestion
```

**Review by gabalafou — COMMENTED** [review]
Okay I double checked the architecture doc page and left some comments/suggestions

File: docs/source/projects/architecture/content-architecture.rst (hunk 9 above)
Comment (isabela-pf) [reply docs/source/projects/architecture/content-architecture.rst:196]: I personally think the best option would be to have this be a part of the main text with an empty alt attribute, but I know I do not have the time to write it at the moment. My current guess is that it would make sense to start by describing the overlapping content first to minimize repetition, then call out areas by section. 

Maybe it's the wrong attitude, but I was going with "something is better than nothing" for this. If it's a blocker, let me know and I'll find a way to get it done.

If we think the image is the way to go, I think this one needs a linked long description. Even that will be tricky with all the relationships!

File: docs/source/projects/architecture/content-architecture.rst (hunk 6 above)
Comment (isabela-pf) [reply docs/source/projects/architecture/content-architecture.rst]: Is there a reason not to change it to a period? Maybe I'm missing something.

**Comment by isabela-pf** [comment]
While there are two open questions left, I think most of this PR has been reviewed. 

I'm finding the following as still open.
- [Questions on the architecture diagram](https://github.com/jupyter/jupyter/pull/607#discussion_r883996291)
- [Question about how punctuation is read](https://github.com/jupyter/jupyter/pull/607#discussion_r883991317)

I want to ping that I think this is, from my perspective, as done as it will be barring further feedback. Do the reviewers agree?

File: docs/source/projects/architecture/content-architecture.rst (hunk 9 above)
Comment (gabalafou) [reply docs/source/projects/architecture/content-architecture.rst:196]: Agreed. This will need to be revisited in the future.

**Comment by isabela-pf** [comment]
Hi there! Just another ping to check on the status of this. Is there any other review that needs to be addressed? 

(I'm taking the 👍 from @gabalafou as a confirmation this PR meets his standards; thanks!)

**Comment by choldgraf** [comment]
I'm sorry! I feel guilty, this has been on my list to follow up on for like a month. I am still hoping to get to it but if somebody else wants to take a pass I'm +1 on merging if others are as well, I don't want to bottleneck this!

**Review by choldgraf — APPROVED** [review]
I could not for the life of me figure out how to get footnotes working in reStructuredText, so I converted that page to MyST markdown and used Markdown footnote syntax. I think this is good to go.

Thanks for the contribution @isabela-pf !

**Comment by isabela-pf** [comment]
Ah, oh my gosh! Thank you for merging, and thanks for your commitment to reviewing this in general! There was no rush; I just leave comments because I know things get buried in my notifications all the time.
